mesonbuild/interpreter.py

```diff
@@ -582,6 +587,9 @@ def __repr__(self):
     def full_path_method(self, args, kwargs):
         return self.interpreter.backend.get_target_filename_abs(self.held_object)
 
+    def index_protocol(self, args, kwargs):
+        return CustomTargetIndexHolder(self.held_object.get_index_protocol(args[0]))
```

Comment:
Same here, can implement `__getitem__`, etc, and return `InterpreterException` for set/del.

Reply:
We also need a `FailureTests` unit test for setting by index.

Reply:
But that would mean implementing `__getitem__` with a non-standard signature. That seems *really* mean.